### ISSUE TITLE
Issue with getItemOffers() not returning [my_offer] and incorrect [is_buy_box_winner] 

### ISSUE THREAD

**Issue by JeffMash** [jlevers/selling-partner-api#471]
Not sure if this is a bug or not.  

I need to find out which listing currently has the Buy Now box.

I have an example ASIN: B008Z20KP0

On that ASIN, my company currently holds the Buy Now box (at $18.96).   There is one competitor with a lower price ($18.75) who doesn't have the Buy Now box.

When I call getItemOffers(), I see that it correctly returns the two offers.   However, I see two problems:

1.  The [my_offer] variable is blank on both.   It should show "1" for my offer since I am the seller making the request.

2.  The [is_buy_box_winner] isn't correct.   It's showing "[is_buy_box_winner] =>1" for the offer that doesn't have the Buy Now box.

But it's showing "[is_buy_box_winner] => " for my listing which IS occupying the Buy Now box.

Are we doing something wrong?

I would love to have the [is_buy_box_winner] variable work correctly, so that way I can exit my function knowing that our company already holds the listing.

**Comment by KhorneHoly** [comment]
You can find out, if the mistake is done on the SP-API or in this library.

You can activate Debug Output in the [Configuration.php](https://github.com/jlevers/selling-partner-api/blob/main/lib/Configuration.php#L195) file.

```
$config = new Config($arrOptions);
$config->setDebug(true);
$config->setDebugFile($pathToWhereYouWantToLocateYourDebugLog);
```
In the file you can find the original request sent to Amazon and the response received from Amazon.
If the error is in the response, the mistake is done in the Amazon SP-API and you must contact the developer support.

**Comment by jlevers** [comment]
This seems unlikely to be an issue with the library, because it _is_ successfully decoding a buybox winner value for one of the offers. It seems more likely to me that the SP API has some lag. I've found the fastest way to get real-time pricing info is via the `ANY_OFFER_CHANGED` notification.

If you provide a dump file as @KhorneHoly suggested, I'll be able to tell for sure if this is an API problem or a library problem.

**Comment by JeffMash** [comment]
Yeah, I think it's SP API lag (not your library).  I would say we can close this issue.

**Comment by JeffMash** [comment]
> I've found the fastest way to get real-time pricing info is via the `ANY_OFFER_CHANGED` notification.

Can you tell me how we can access the AnyOfferChangedNotification?   Example code if possible.

(I am already subscribed to the ANY_OFFER_CHANGED event type .   I can even see messages in the Amazon SQS --> Queue.    I just don't know what function in your API to retrieve and enumerate through the messages.)

Can you point me in the right direction?
